utils/files.ts

```diff
@@ -98,3 +118,165 @@ export function streamToCSV (
+      rate,
+      currency,
+      address: DEFAULT_MULTI_VALUES_LINE_LABEL,
+      newtworkId: tempGroup[0].address.networkId,
+      notes
+    } as TransactionFileData)
+
+    tempGroups[groupKey] = []
+  }
+
+  const pushTx = (tx: TransactionsWithPaybuttonsAndPrices): void => {
+    const { timestamp, hash, address, amount } = tx
+    const values = getTransactionValue(tx)
+    const value = Number(values[currency])
+    const rate = value / Number(amount)
+
+    treatedPayments.push({
+      amount,
+      value,
+      date: moment.tz(timestamp * 1000, timezone),
+      transactionId: hash,
+      rate,
+      currency,
+      address: address.address,
+      notes: '',
+      newtworkId: address.networkId
+    } as TransactionFileData)
+    totalPaymentsTreated += 1
+  }
+
+  payments.forEach((tx: TransactionsWithPaybuttonsAndPrices, index: number) => {
+    const { timestamp } = tx
+    const values = getTransactionValue(tx)
+    const value = Number(values[currency])
+    const dateKey = moment.tz(timestamp * 1000, timezone).format('YYYY-MM-DD')
+    const dateKeyUTC = moment.utc(timestamp * 1000).format('YYYY-MM-DD')
+    const groupKey = `${dateKey}_${dateKeyUTC}`
+
+    const nextPayment = payments[index + 1]
+    const nextDateKey = nextPayment === undefined ? null : moment.tz(nextPayment.timestamp * 1000, timezone).format('YYYY-MM-DD')
+    const nextDateKeyUTC = nextPayment === undefined ? null : moment.utc(nextPayment.timestamp * 1000).format('YYYY-MM-DD')
+    const nextGroupKey = nextDateKey === null || nextDateKeyUTC === null ? null : `${nextDateKey}_${nextDateKeyUTC}`
+
+    if (value < collapseThreshold) {
+      if (tempGroups[groupKey] === undefined) tempGroups[groupKey] = []
+      tempGroups[groupKey].push(tx)
+    } else {
+      Object.keys(tempGroups).forEach(pushTempGroup)
+      pushTx(tx)
+    }
+
+    if (nextGroupKey !== groupKey) {
+      pushTempGroup(groupKey)
+    }
+  })
+
+  Object.keys(tempGroups).forEach(pushTempGroup)
+
+  if (totalPaymentsTreated !== payments.length) {
+    throw new Error('Error to collapse payments')
+  }
+
+  return treatedPayments
+}
+
+const sortPaymentsByNetworkId = (payments: TransactionsWithPaybuttonsAndPrices[]): TransactionsWithPaybuttonsAndPrices[] => {
+  const groupedByNetworkIdPayments = payments.reduce<Record<number, TransactionsWithPaybuttonsAndPrices[]>>((acc, transaction) => {
+    const networkId = transaction.address.networkId
+    if (acc[networkId] === undefined || acc[networkId] === null) {
+      acc[networkId] = []
+    }
+    acc[networkId].push(transaction)
+    return acc
+  }, {})
+
+  return Object.values(groupedByNetworkIdPayments).reduce(
+    (acc, curr) => acc.concat(curr),
+    []
+  )
+}
+
+const getPaybuttonTransactionsFileData = (transactions: TransactionsWithPaybuttonsAndPrices[], currency: SupportedQuotesType, timezone: string): TransactionFileData[] => {
+  const paymentsFileData: TransactionFileData[] = []
+  transactions.forEach(element => {
```

Comment:
`element` is a bad name for what you know is a transaction. Any array has elements, this is the variable-naming equivalent of typing something as `any`. Just call it `tx` or `transaction`.